src/client/app/utils/api/ApiBackend.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import axios, {AxiosRequestConfig, AxiosResponse} from 'axios';
+import {getToken, hasToken} from '../token';
+
+/**
+ * Handles the actual sending and receiving of network requests.
+ *
+ * Should probably only be used by other api classes.
+ */
+class ApiBackend {
+	public async doGetRequest<R>(
+		url: string,
+		params: { [key: string]: string } = {},
+		headers: { [key: string]: string } = {},
+		extraConfig: AxiosRequestConfig = {}
+	): Promise<R> {
+		const response: AxiosResponse = await axios.get(url, this.buildConfig(params, headers, extraConfig));
+		return response.data as R;
+	}
+
+	public async doPutRequest<R>(
+		url: string,
+		body: any,
+		params: { [key: string]: string } = {},
+		headers: { [key: string]: string } = {},
+		extraConfig: AxiosRequestConfig = {}
+	): Promise<R> {
+		const response: AxiosResponse = await axios.put(url, body, this.buildConfig(params, headers, extraConfig));
+		return response.data as R;
+	}
+
+	public async doPostRequest<R>(
+		url: string,
+		body: any,
+		params: { [key: string]: string } = {},
+		headers: { [key: string]: string } = {},
+		extraConfig: AxiosRequestConfig = {}
+	): Promise<R> {
+		const response: AxiosResponse = await axios.post(url, body, this.buildConfig(params, headers, extraConfig));
+		return response.data as R;
+	}
+
+	private buildConfig(
+		params: { [key: string]: string},
+		headers: { [key: string]: string},
+		extraConfig: AxiosRequestConfig
+	): AxiosRequestConfig {
+		if (hasToken()) {
```

Comment:
Doesn't this always send a token regardless of the route? Is this best practice?

Reply:
It does. I think this is a good way to do it. It makes things available to the server more often and there's less of a downside. We can just make the server send a 401 unauthorized if the user doesn't have the appropriate authorization level.